skxray/diff_roi_choice.py

```diff
@@ -0,0 +1,354 @@
+# * Redistributions of source code must retain the above copyright     #
+#   notice, this list of conditions and the following disclaimer.      #
+#                                                                      #
+# * Redistributions in binary form must reproduce the above copyright  #
+#   notice this list of conditions and the following disclaimer in     #
+#   the documentation and/or other materials provided with the         #
+#   distribution.                                                      #
+#                                                                      #
+# * Neither the name of the Brookhaven Science Associates, Brookhaven  #
+#   National Laboratory nor the names of its contributors may be used  #
+#   to endorse or promote products derived from this software without  #
+#   specific prior written permission.                                 #
+#                                                                      #
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS  #
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT    #
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS    #
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE       #
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,           #
+# INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES   #
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR   #
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)   #
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,  #
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OTHERWISE) ARISING   #
+# IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE   #
+# POSSIBILITY OF SUCH DAMAGE.                                          #
+########################################################################
+
+"""
+This module is to get informations of different region of interests(roi's).
+ Information : the number of pixels, pixel list, indices 
+"""
+
+
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+import six
+
+from six.moves import zip
+from six import string_types
+
+import time
+import sys
+import numpy as np
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def roi_rectangles(num_rois, roi_data, detector_size):
+    """
+    Parameters
+    ----------
+    num_rois: int
+        number of region of interests(roi)
+
+    roi_data: ndarray
+        upper left co-ordinates of roi's and the, length and width of roi's
+        from those co-ordinates
+        shape is [num_rois][4]
+
+    detector_size : tuple
+        2 element tuple defining the number of pixels in the detector.
+        Order is (num_rows, num_columns)
+
+    Returns
+    -------
+    roi_inds : ndarray
+        indices of the required shape
+        shape [detector_size[0]*detector_size[1]][1]
```

Comment:
Is this really `(N, 1)` or `(N,)`.  I would suggest using notation that matches what `a.shape` returns.